client/components/domains/search-filters/tld-filter-bar.jsx

```diff
@@ -0,0 +1,205 @@
+		onSubmit: PropTypes.func.isRequired,
+		query: PropTypes.string.isRequired,
+		recordTldFilterSelected: PropTypes.func.isRequired,
+		showPlaceholder: PropTypes.bool.isRequired,
+	};
+
+	static defaultProps = {
+		numberOfTldsShown: 8,
+	};
+
+	state = {
+		showPopover: false,
+	};
+
+	bindButton = button => ( this.button = button );
+
+	handleButtonClick = event => {
+		const isCurrentlySelected = event.currentTarget.dataset.selected === 'true';
+		const newTld = event.currentTarget.value;
+
+		this.props.recordTldFilterSelected(
+			newTld,
+			event.currentTarget.dataset.index,
+			! isCurrentlySelected
+		);
+
+		const tldSet = new Set( [ ...this.props.filters.tlds, newTld ] );
+		if ( isCurrentlySelected ) {
+			tldSet.delete( newTld );
+		}
+
+		const tlds = [ ...tldSet ].filter( tld => includes( this.props.availableTlds, tld ) );
+		this.props.onChange( { tlds }, { shouldSubmit: true } );
+	};
+	handleFiltersReset = () => {
+		this.togglePopover();
+		this.props.onFiltersReset( 'tlds' );
+	};
+	handleFiltersSubmit = () => {
+		this.togglePopover();
+		this.props.onFiltersSubmit();
+	};
+	handleTokenChange = newTlds => {
+		this.props.onChange(
+			{ tlds: newTlds.filter( tld => includes( this.props.availableTlds, tld ) ) },
+			{ shouldSubmit: true }
+		);
+	};
+
+	togglePopover = () => {
+		this.setState( {
+			showPopover: ! this.state.showPopover,
+		} );
+	};
+
+	getSortedAvailableTlds() {
+		const { availableTlds, query } = this.props;
+		// NOTE: If TLD is part of the user query, make it 5x more important than
+		//       the original order of the availableTlds array.
+		//
+		//       The comparison value is set to negative to force a descending sort.
+		return sortBy(
+			availableTlds,
+			tld =>
+				-(
+					5 * includes( query, tld ) +
+					( 1 - availableTlds.indexOf( tld ) / availableTlds.length )
+				)
+		);
+	}
+
+	render() {
+		const isKrackenUi = config.isEnabled( 'domains/kracken-ui/filters' );
+		if ( ! isKrackenUi ) {
+			return null;
+		}
+
+		if ( this.props.showPlaceholder ) {
+			return this.renderPlaceholder();
+		}
+
+		const {
+			filters: { tlds },
+			lastFilters: { tlds: selectedTlds },
+			translate,
+			numberOfTldsShown,
+		} = this.props;
+		const hasFilterValue = tlds.length > 0;
+
+		return (
+			<CompactCard className="search-filters__buttons">
+				{ this.getSortedAvailableTlds()
+					.slice( 0, numberOfTldsShown )
+					.map( ( tld, index ) => (
+						<Button
+							className={ classNames( { 'is-active': includes( selectedTlds, tld ) } ) }
+							data-selected={ includes( selectedTlds, tld ) }
+							data-index={ index }
+							key={ tld }
+							onClick={ this.handleButtonClick }
+							value={ tld }
+						>
+							.{ tld }
+						</Button>
+					) ) }
+				<Button
+					className={ classNames( { 'is-active': hasFilterValue } ) }
+					onClick={ this.togglePopover }
+					ref={ this.bindButton }
+					key="popover-button"
+				>
+					{ translate( 'More Extensions', {
```

Comment:
Hi! I've found a possible matching string that has already been translated **1** time:
`translate( 'Extensions', { context: 'Refers to top level domain name extension, such as ".com"'} )` *ES Score: 19*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

:information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

Reply:
Fixed. Now uses both `context` and `comment`.